### PR TITLE
Develop/refactor

### PR DESCRIPTION
**Pull request by thecozies**
## goals
- increase performance where changes were questionable
- goals that conflicted with each other:
    - decrease number of changes from base decomp
    - decrease number of changes from HackerSM64 master
    - retain documentation and variable naming
    - altered formatting changes that made code difficult to read:
        - excessive ternaries
        - excessive parenthesis
        - restore whitespace that was removed
        - multiline if statements changed to be aligned from the && instead of after the &&
- remove config options that seemed to have no value
- reassess config naming conventions
- identify (and fix) HackerSM64 bugs
- reimplement non-detrimental bugs used by speedrunners
- some code documentation to explain hackersm64 changes
    - i wish i was more adamant about this, but in some places I added `//! HackerSM64: message` so that changes were explained and clear

___

## file specific notes
- ingame_menu
    - had SO many changes, i left all new changes
    - I tried to revert it but the old file broke a lot of other dependencies that had changed
- macro_special_objects
    - same as ingame_menu, wasn't worth my time due to all of the changes, and also most people avoid macro objects in hacks anyways
- mario actions automatic
    - changed `NO_FALSE_LEDGEGRABS` to `LEDGE_GRABS_CHECK_SLOPE_ANGLE` because it allows ledge grabs on walls under slippery slopes, which is a false ledgegrab

___

## questions
- `BETTER_*` named config options
    - these are pretty opinionated at times - does it make more sense to instead call these `NEW_*` or something?
    - maybe some should be split into various parameters
    - for example, `BETTER_HANGING` config option
        - hanging doesnt require holding A, instead requires you to press A or B to let go
        - imo, i think the way hanging works by holding A makes more sense but theres other aspects that are nice
            - it has increased speed
            - better turning
            - accelerated speed based on stick mag
- object config options
    - i believe that some of these could be in the specific file, or just always default
    - config options should be concise and important - many of those config opts are minor bug fixes
- `GROUND_TURN_FIX`
    - This is so incredibly different from vanilla i dont think we can call this a fix
    - what should this be named? is it worth having as an option? some code is commented out and this area is very messy

___

## What to look out for when reviewing
- Do any changes seem like they could hurt performance?
- Do any changes make the code less legible?
- Do any changes seem like they would cause major, or impossible to resolve conflicts to those updating?
- Which config options should be default?

### CODE REVIEW

**Comment by thecozies** [comment]
# Note: Post all bugs here!

https://github.com/Reonu/HackerSM64/projects/1?add_cards_query=is%3Aopen

**Comment by Reonu** [comment]
>     * `GROUND_TURN_FIX`
>       
>       * This is so incredibly different from vanilla i dont think we can call this a fix
>       * what should this be named? is it worth having as an option? some code is commented out and this area is very messy

I fully agree. It's not a fix, it doesn't fix any problems, it simply fundamentally changes the physics and the way the game is played. Its existence is warranted since some people really like it, but it should NOT be called a fix and obviously should never ever become the default.

File: include/config/config_movement.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+/*********************
+ * MOVEMENT SETTINGS *
+ *********************/
+
+// Fixes Mario's turn ground radius by making it dependent on the analog stick magnitude.
+//#define GROUND_TURN_FIX
+
+// Fixes Mario's turn ground radius by allowing Mario to turn around at any speed.
+// Basically a simpler version of GROUND_TURN_FIX but smoother & closer to vanilla.
+#define GROUND_TURNING_AROUND_FIX
```
Comment (Reonu) [inline include/config/config_movement.h:12]:
```suggestion
// Fixes Mario's turn ground radius by making it dependent on the analog stick magnitude.
//#define INSTANT_GROUND_TURN

// Fixes Mario's turn ground radius by allowing Mario to turn around at any speed.
// Basically a simpler version of INSTANT_GROUND_TURN but smoother & closer to vanilla.
//#define INSTANT_GROUND_TURN_LITE
```
- This is not a fix. There is no problem being fixed here. It changes the physics of the game to something different. So it should not be called a "fix".
- Please don't turn these on by default

**Review by Reonu — COMMENTED** [review]
Does this fix the game crashng when grabbing stuff when rumble is on? Or is that a separate issue?

**Review by Reonu — APPROVED** [review]
lgtm

**Comment by thecozies** [comment]
Closed for release PR #276

**Pull request closed without merging**